### PR TITLE
feat: meme command

### PR DESCRIPTION
**Pull request by nsylke**
### Items Changed

- [X] Commands
- [ ] Events
- [ ] Handlers
- [ ] Structures
- [ ] Tasks
- [ ] Documentation
- [ ] Other: ______

### Description

Meme command

### Have You...?

- [X] I have read the [Contributing Guidelines](https://github.com/typicalbot/typicalbot/blob/master/.github/CONTRIBUTING.md).
- [X] I have completed the [pull request template](https://github.com/typicalbot/typicalbot/blob/master/.github/PULL_REQUEST_TEMPLATE.md).
- [X] I have checked open [pull requests](https://github.com/typicalbot/typicalbot/pulls) for upcoming features and fixes.

### CODE REVIEW

**Comment by tjrgg** [comment]
What about a parameter for searching for a meme by query and then if no query is specified, pull a random one from top?

**Comment by nsylke** [comment]
> What about a parameter for searching for a meme by query and then if no query is specified, pull a random one from top?

Won't be in this PR. I would suggest creating that as a new issue (enhancement).

**Comment by tjrgg** [comment]
> > What about a parameter for searching for a meme by query and then if no query is specified, pull a random one from top?
> 
> Won't be in this PR. I would suggest creating that as a new issue (enhancement).

I'd prefer to see it together with this PR and not merge this without it.

**Review by tjrgg — CHANGES_REQUESTED** [review]
The command doesn't have a non-embed response.

File: src/commands/fun/meme.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { TypicalGuildMessage } from '../../lib/types/typicalbot';
+import { MessageEmbed } from 'discord.js';
+
+export default class extends Command {
+    async execute(message: TypicalGuildMessage) {
+        const { data: { children } } = await fetch('https://www.reddit.com/r/memes/top.json?sort=top&t=day&limit=500')
+            .then(res => res.json());
+
+        const raw = children.filter((c: any) => c.data.over_18 === false);
+        const meme = raw[Math.floor(Math.random() * raw.length)].data;
```
Comment (Skillz4Killz) [inline src/commands/fun/meme.ts:16]: This seems like a potential bug to me. What happens if for whatever reason, the filter makes this empty ie all of over 18. You would have `undefined.data` which would throw an error

Comment (nsylke) [reply src/commands/fun/meme.ts:16]: You are definitely correct. I have resolved this.

**Review by Skillz4Killz — APPROVED** [review]

**Review by tjrgg — APPROVED** [review]

**Pull request merged**